src/OrleansEventSourcing/EventStorage/IEventStore.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Orleans.Providers;
+
+namespace Orleans.EventSourcing.EventStorage
+{
+   
+    /// <summary>
+    /// The event store interface, as implemented by event storage providers
+    /// </summary>
+    public interface IEventStore
+    {
+        /// <summary>
+        /// Loads the sequence of events in an event stream
+        /// </summary>
+        /// <param name="streamName">The name of the event stream</param>
+        /// <param name="startAtVersion">The version at which to start</param>
+        /// <param name="endAtVersion">The version at which to end, or null to return up and including the most recent</param>
+        /// <returns>A structure containing a segment of the event sequence</returns>
+        Task<EventStreamResponse> LoadStream(string streamName, int startAtVersion = 0, int? endAtVersion = null);
+
+        /// <summary>
+        /// Stores a sequence of events in an event stream
+        /// </summary>
+        /// <param name="streamName">The name of the event stream</param>
+        /// <param name="events">The sequence of events, including their Guids</param>
+        /// <param name="expectedVersion">null for unconditional events, or the expected version for conditional events</param>
+        /// <returns>a boolean task that completes when the append is confirmed, and returns true if there was no version mismatch</returns>
+        Task<bool> AppendToStream(string streamName, IEnumerable<KeyValuePair<Guid, object>> events, int? expectedVersion = null);
```

Comment:
Yes, it is a guid for each event. Its purpose is to avoid duplicate appends when retrying a failed append. 

This same pattern is already used by both GetEventStore and StreamStone, so adding a Guid to this interface does not really change what is in storage. 

The grain code does not have to specify the guids, but can do so optionally (by implementing IEventWithGuid on the eventclass).